hm_pyhelper/miner_param.py

```diff
@@ -59,9 +59,14 @@ def provision_key():
     direct_path = os.path.dirname(os.path.abspath(__file__))
     gateway_mfr_path = os.path.join(direct_path, 'gateway_mfr')
 
-    if get_gateway_mfr_test_result():
-        logging.info("Key already provisioned")
-        return True
+    test_results = get_gateway_mfr_test_result()
+    if test_results['result'] == 'pass':
+        for test in test_results:
```

Comment:
The second reason is that we want to iterate over the `tests` array but this is iterating a level higher over the whole response object.